src/main/java/org/spongepowered/api/entity/living/player/User.java

```diff
@@ -70,6 +72,33 @@
      */
     Optional<Player> getPlayer();
 
+    /**
+     * Gets the position of this User
+     *
+     * @return The position of this User
+     */
+    Vector3d getPosition();
+
+    /**
+     * Gets the World UUID of this User.
+     *
+     * <p>May return empty when the world the player is in does not exist anymore</p>
+     *
+     * @return The World UUID of this User if found
+     */
+    Optional<UUID> getWorldUniqueId();
+
+    /**
+     * Sets the position and world of this User.
+     * <p>The UUID must belong to an existing world.</p>
+     *
+     * @param position The position to set
+     * @param world The world UUID to set
+     *
+     * @throws IllegalArgumentException When the UUID does not belong to an existing world.
+     */
+    void setLocation(Vector3d position, UUID world);
```

Comment:
Maybe we should name this `setPosition` to match the `getPosition`? Since it is really the position and not our version of "location".

Reply:
I chose the name because it also sets the world.
For online players this will literally call `setLocation(Location)`

Reply:
Ah yeah I didn't consider that, makes sense to me.

Reply:
I wonder if there is a way to remove the position of a user, i.e. to "unstuck" it

Reply:
What do you consider as unstuck?
Teleport him back to spawn?

Maybe we should state in the setters that they will delegate to Player#setX if the player is online

Reply:
You can access player bedspawn and world spawn. Just set it accordingly.